index.d.ts

```diff
@@ -10,6 +10,9 @@ declare module 'fastify' {
     > {
     ip: string | number
   }
+  interface FastifyInstance {
+    rateLimit: preHandlerAsyncHookHandler;
+  }
```

Comment:
I haven't tested this but believe it's accurate.

Reply:
A proper test of the `rateLimit()` decorator used on the `setNotFoundHandler` method will require a fix on fastify to expose the optional hooks to the method. Cf. https://github.com/fastify/fastify/pull/2929

Reply:
Inside https://github.com/fastify/fastify-rate-limit/blob/master/test/types/types.test.ts just call the function and add an assertion to validate the returned type.